python/ctsm/modify_fsurdat/fsurdat_modifier.py

```diff
@@ -0,0 +1,134 @@
+        file_path=args.cfg_path, is_list=True, convert_to_type=float)
+    sai = get_config_value(config=config, section=section, item='sai',
+        file_path=args.cfg_path, is_list=True, convert_to_type=float)
+    hgt_top = get_config_value(config=config, section=section,
+        item='hgt_top', file_path=args.cfg_path, is_list=True,
+        convert_to_type=float)
+    hgt_bot = get_config_value(config=config, section=section,
+        item='hgt_bot', file_path=args.cfg_path, is_list=True,
+        convert_to_type=float)
+
+    soil_color = get_config_value(config=config, section=section,
+        item='soil_color', file_path=args.cfg_path,
+        allowed_values=['1','2','3','4','5','6','7','8','9','10','11',
+        '12','13','14','15','16','17','18','19','20',CONFIG_UNSET],
+        convert_to_type=int)
+
+    std_elev = get_config_value(config=config, section=section,
+        item='std_elev', file_path=args.cfg_path, convert_to_type=float)
+    max_sat_area = get_config_value(config=config, section=section,
+        item='max_sat_area', file_path=args.cfg_path,
+        convert_to_type=float)
+
+    # Create ModifyFsurdat object
+    modify_fsurdat = ModifyFsurdat(fsurdat_in, lon_1=lnd_lon_1,
+        lon_2=lnd_lon_2, lat_1=lnd_lat_1, lat_2=lnd_lat_2,
+        landmask_file=landmask_file)
+
+    # ------------------------------
+    # modify surface data properties
+    # ------------------------------
+
+    # Set fsurdat variables in a rectangle that could be global (default).
+    # Note that the land/ocean mask gets specified in the domain file for
+    # MCT or the ocean mesh files for NUOPC. Here the user may specify
+    # fsurdat variables inside a box but cannot change which points will
+    # run as land and which as ocean.
+    if idealized:
+        modify_fsurdat.set_idealized()  # set 2D variables
+        # set 3D and 4D variables pertaining to natural vegetation
+        modify_fsurdat.set_dom_nat_pft(dom_nat_pft=0, lai=[], sai=[],
+                                       hgt_top=[], hgt_bot=[])
+
+    if dom_nat_pft is not None:  # overwrite "idealized" value
+        modify_fsurdat.set_dom_nat_pft(dom_nat_pft=dom_nat_pft,
+                                       lai=lai, sai=sai,
+                                       hgt_top=hgt_top, hgt_bot=hgt_bot)
+
+    if max_sat_area is not None:  # overwrite "idealized" value
+        modify_fsurdat.file['FMAX'] = \
+         modify_fsurdat.file['FMAX'].where(modify_fsurdat.not_rectangle, other=max_sat_area)
+
+    if std_elev is not None:  # overwrite "idealized" value
+        modify_fsurdat.file['STD_ELEV'] = \
+         modify_fsurdat.file['STD_ELEV'].where(modify_fsurdat.not_rectangle, other=std_elev)
+
+    if soil_color is not None:  # overwrite "idealized" value
+        modify_fsurdat.file['SOIL_COLOR'] = \
+         modify_fsurdat.file['SOIL_COLOR'].where(modify_fsurdat.not_rectangle, other=soil_color)
```

Comment:
(A good general rule in object-oriented design is that all member variables of a class should be private, only accessed within functions defined in that class. This helps long-term maintainability and understandability of the code. As with any rules, there are times when it's worth breaking the rule; for example, we break this rule all over the place in CTSM, for performance reasons and because the overall design of CTSM doesn't lend itself to following this rule. But this rule is generally a good starting point for object-oriented design.)

Reply:
I have addressed this now.

Reply:
Thanks, this is moving towards what I imagined but still could be improved for a more standard object-oriented design: Instead of code like:

```python
modify_fsurdat.file['FMAX'] = modify_fsurdat.setvar(
            modify_fsurdat.file['FMAX'], max_sat_area)
```

replace this with

```python
modify_fsurdat.setvar('FMAX', max_sat_area)
```

so that `file` becomes a private variable of this class that isn't modified outside of it.

You can let me know if you see a problem with this approach.

Reply:
Thanks, @billsacks 
I had originally tried what you're suggesting here but opted for a way to make the function shorter. Now I'm trying your suggestion, which works. Let me know if you see an elegant way to shorten the function in this context.

Reply:
Scrolling through your comments, I see that you have further suggestions regarding this function which may answer my question.